doc/reference/javascript_reference.rst

```diff
@@ -139,71 +151,223 @@ and if necessary, will create/recreate the corresponding bundles.
+directive:
+
+.. code-block:: py
 
-    <template id="assets_backend" name="helpdesk assets" inherit_id="web.assets_backend">
-        <xpath expr="//script[last()]" position="after">
-            <link rel="stylesheet" type="text/scss" href="/helpdesk/static/src/scss/helpdesk.scss"/>
-            <script type="text/javascript" src="/helpdesk/static/src/js/helpdesk_dashboard.js"></script>
-        </xpath>
-    </template>
+    'assets_common': [
+        ('remove', 'web/static/src/js/boot.js'),
+    ],
 
+g) Replace an asset file with one or multiple file(s): replace
+
+Let us now say that an asset need not only to be removed, but you also want to insert
+your new version of that asset at the same exact position. This can be done with
+the *replace* directive, using a 3-element tuple `('replace', target, asset)`:
+
+.. code-block:: py
+
+    'assets_common': [
+        ('replace', 'web/static/src/js/boot.js', 'my_addon/static/src/js/boot.js'),
+    ],
+
+Note that directives targetting a certain file (i.e. *before*, *after*, *replace* and *remove*)
+need that file to be declared beforehand, either in manifests higher up in the hierarchy
+or in 'ir.asset' records with a lower sequence.
 
 .. note ::
 
     Note that the files in a bundle are all loaded immediately when the user loads the
-    odoo web client.  This means that the files are transferred through the network
-    every time (except when the browser cache is active).  In some cases, it may be
-    better to lazyload some assets.  For example, if a widget requires a large
+    odoo web client. This means that the files are transferred through the network
+    every time (except when the browser cache is active). In some cases, it may be
+    better to lazyload some assets. For example, if a widget requires a large
     library, and that widget is not a core part of the experience, then it may be
     a good idea to only load the library when the widget is actually created. The
     widget class has actually builtin support just for this use case. (see section
     :ref:`reference/javascript_reference/qweb`)
 
+Assets loading order
+-------------------
+
+The order in which assets are loaded is sometimes critical and must be deterministic,
+mostly for stylesheets priorities and setup scripts. Assets in Odoo are processed
+as following:
+
+1. When an asset bundle is called (e.g. `t-call-assets="assets_common"`), an empty
+list of assets is generated
+
+2. All records of type 'ir.asset' matching the bundle will be fetched and sorted
+by sequence number. Then all records with a sequence lower than 16 will be processed
+and applied to the current list of assets.
+
+3. All addons declaring assets for said bundle will apply their assets operations
+to this list. This is done following the order of modules dependencies (e.g. 'web'
+assets will be read before 'website'). Any directive trying to add a file already
+present in the list will be ignored.
```

Comment:
All addons declaring assets for said bundle **in their manifest** will apply...

"Any directive trying to add a file already present in the list will be ignored."
Strictly speaking, that says that the directive `my_addon/static/lib/jquery/**/*` is ignored (as a whole) because `my_addon/static/lib/jquery/jquery.js` is already in the list.
What you mean is: "In case a file is added several times to the list, only it first occurrence is kept."